connection-cache/src/connection_cache.rs

```diff
@@ -583,7 +582,14 @@ mod tests {
 
                 let conn = &map.get(addr).expect("Address not found").get(0).unwrap();
                 let conn = conn.new_blocking_connection(*addr, connection_cache.stats.clone());
-                assert!(addr.ip() == conn.server_addr().ip());
+                assert_eq!(
+                    BlockingClientConnection::server_addr(&*conn).ip(),
+                    addr.ip(),
+                );
+                assert_eq!(
```

Comment:
Isn't conn blocking connection?

Reply:
It implements both interfaces. Does not hurt to assert both.